From 5d4ad66bdde2ff103c848dc2709a5a67f4d2844f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 22 Aug 2017 13:25:25 +0300
Subject: [PATCH] Fix cross compilation of compiler-rt for i386 mingw64

---
This patch hasn't been sent for review anywhere yet - the real issue hasn't
been analyzed properly yet.
---
 cmake/base-config-ix.cmake | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/cmake/base-config-ix.cmake b/cmake/base-config-ix.cmake
index b38c6ca..103bca0 100644
--- a/cmake/base-config-ix.cmake
+++ b/cmake/base-config-ix.cmake
@@ -134,19 +134,19 @@ macro(test_targets)
     if(COMPILER_RT_DEFAULT_TARGET_ONLY)
       add_default_target_arch(${COMPILER_RT_DEFAULT_TARGET_ARCH})
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "i[2-6]86|x86|amd64")
-      if(NOT MSVC)
-        test_target_arch(x86_64 "" "-m64")
-        # FIXME: We build runtimes for both i686 and i386, as "clang -m32" may
-        # target different variant than "$CMAKE_C_COMPILER -m32". This part should
-        # be gone after we resolve PR14109.
-        test_target_arch(i686 __i686__ "-m32")
-        test_target_arch(i386 __i386__ "-m32")
-      else()
+#      if(NOT MSVC)
+#        test_target_arch(x86_64 "" "-m64")
+#        # FIXME: We build runtimes for both i686 and i386, as "clang -m32" may
+#        # target different variant than "$CMAKE_C_COMPILER -m32". This part should
+#        # be gone after we resolve PR14109.
+#        test_target_arch(i686 __i686__ "-m32")
+#        test_target_arch(i386 __i386__ "-m32")
+#      else()
         if (CMAKE_SIZEOF_VOID_P EQUAL 4)
           test_target_arch(i386 "" "")
         else()
           test_target_arch(x86_64 "" "")
-        endif()
+#        endif()
       endif()
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "powerpc")
       TEST_BIG_ENDIAN(HOST_IS_BIG_ENDIAN)
-- 
2.7.4

